flake.nix

```diff
@@ -0,0 +1,54 @@
+{
+  description = "Nix flake for Git, Node.js, TypeScript, FNM, and Redis";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          config = {
+            allowUnfree = true;
+          };
+        };
+      in
+      {
+        devShells.default = pkgs.mkShell {
+          buildInputs = [
+            pkgs.git
+            pkgs.nodejs
+            pkgs.typescript
+            pkgs.fnm
+            pkgs.redis
+          ];
+
+          shellHook = ''
```

Comment:
Don't use a shellHook. This is what is creating the unnecessary overhead. You don't need an ephemeral shell to perform these actions; they should be done in the respective build phases.